### PR TITLE
Update search component to input configurations through searchSettings

### PR DESCRIPTION
**Pull request by cwolf10**
## Description
Search component configurations should be inputed through searchSettings rather than a separate input for each field.

**Note: Potential breaking change**
If MFE is taking advantage of inputClass or parentSelector, input values will need to be updated.

## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [ ] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [x] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Screenshots (if appropriate):
Screen Shot 2020-12-04 at 3 49 13 PM

## Which browsers have you tested?
- [ ] Internet Explorer 11
- [ ] Edge
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

## Checklist:

- [ ] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [ ] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/documentation/src/lib/components/search/demos/optional/search-optional.component.html [hunk 1]
```diff
@@ -1,25 +1,25 @@
 <div>
     <h4>Search with dropdown</h4>
     <p> <code> dropdown: {{'{'}} options: []{{' }'}}</code> is used for selecting search with dropdown mode.</p>
-    <sds-search [(ngModel)]="ddmodel" [searchSettings]="ddSearchSettings" placeholder="Entity an entity ID, name, or keyword"></sds-search>
```
Comment (Bibek-Regmi) [inline libs/documentation/src/lib/components/search/demos/optional/search-optional.component.html:4]: Does the ts file already have the same placeholder as the one used here? If not, could you update the settings in the ts file as well.

File: libs/documentation/src/lib/components/subheader/demos/search-sub-pages/subheader-search-sub-pages.component.ts [hunk 2]
```diff
@@ -89,6 +89,15 @@ export class SubheaderSearchSubPagesComponent {
   options: FormlyFormOptions;
   downloadResponse = {};
 
+  searchSettings: SearchSettings = {
+    placeholder: 'Entity an entity ID, name, or keyword',
```
Comment (Bibek-Regmi) [inline libs/documentation/src/lib/components/subheader/demos/search-sub-pages/subheader-search-sub-pages.component.ts:93]: minor typo here on placeholder

File: libs/documentation/src/lib/components/subheader/demos/search/subheader-search.component.ts [hunk 3]
```diff
@@ -90,6 +90,15 @@ export class SubheaderSearchComponent {
   options: FormlyFormOptions;
   downloadResponse = {};
 
+  searchSettings: SearchSettings = {
+    placeholder: 'Entity an entity ID, name, or keyword',
```
Comment (Bibek-Regmi) [inline libs/documentation/src/lib/components/subheader/demos/search/subheader-search.component.ts:94]: small typo on the word `Enter`

File: libs/documentation/src/lib/components/subheader/demos/tier-2-workspace/subheader-tier-2-workspace.component.ts [hunk 4]
```diff
@@ -94,6 +91,15 @@ export class SubheaderTier2WorkspaceComponent {
   options: FormlyFormOptions;
   downloadResponse = {};
 
+  searchSettings: SearchSettings = {
+    placeholder: 'Entity an entity ID, name, or keyword',
```
Comment (Bibek-Regmi) [inline libs/documentation/src/lib/components/subheader/demos/tier-2-workspace/subheader-tier-2-workspace.component.ts:95]: typo

File: libs/documentation/src/lib/components/search/demos/optional/search-optional.component.html (hunk 1 above)
Comment (cwolf10) [reply libs/documentation/src/lib/components/search/demos/optional/search-optional.component.html:4]: The ts has a different placeholder value, but it is the value that has always shown in the component. The one in the HTML was never patched through, and only visible when examining the HTML.

**Review by Bibek-Regmi — APPROVED** [review]

**Pull request merged**